### PR TITLE
Issue 15 build solution class

### LINKED ISSUE #15: Build Solution Class
- [x] implement add( model1, proto1 ) function
- [x] implement list() function
- [x] implement remove ( idx ) function
- [x] implement visualise ( plt_params ) function
- [x] linting issues
- [x] method documentation
- [x] ODE_system method
- [x] solution method
- Takes Model object and Protocol object as input, implements ODE solvers
- Outputs the solution as Matplotlib plots and numpy arrays, accessible via different methods
- User can specify if they want multiple side-by-side plots, or as overlaid plot
- Can implement multiple functions depending on number of solutions being compared
- Plot central compartment only

### PR DESCRIPTION
**Pull request by jewh**
Finalised working version of solution.py, updated for changes to protocol.py.

Closes #15

### CODE REVIEW

**Comment by jewh** [comment]
Linting issues in Protocol.py should be fixed by merge with issue-67-, and surely the linting issues in test_solution.py are fixed once merged with Master?

File: pkmodel/tests/test_solution.py [hunk 1]
```diff
@@ -10,6 +10,77 @@ def test_create(self):
...
+        Test the solution method in the Solution class.
+        """
+        solution1 = pk.Solution()
+        solution2 = pk.Solution()
+        model1 = pk.Model('sc')
+        protocol1 = pk.Protocol()
+        model2 = pk.Model('iv')
+        protocol2 = pk.Protocol(initial_dose=2, time_span=1.3)
+        solution1.add(model1, protocol1)
+        solution2.add(model2, solution2)
```
Comment (smf541) [inline pkmodel/tests/test_solution.py:80]: should line 80 read "solution2.add(model2, protocol2)"? Then the variable protocol2 is used and the linter won't complain

**Review by smf541 — CHANGES_REQUESTED** [review]
I think that changing line 80 as commented below will let us pass the linting test, but the code coverage test fails because it isn't able to import pkmodel/numpy... I'm not sure how to fix this

File: pkmodel/tests/test_solution.py (hunk 1 above)
Comment (smf541) [reply pkmodel/tests/test_solution.py:80]: or if you are purposefully passing it solution2.add() a wrong input, can we get rid of protocol2 altogether?

Comment (alisonfarrar) [reply pkmodel/tests/test_solution.py]: Fixed this, it should be adding protocol2.

**Comment by smf541** [comment]
I will try adding a line to install requirements.txt to a github actions workflow file to see if tests will run then! 
Currently lines 28-29 read: 
        python -m pip install --upgrade pip
        pip install flake8
I'll change this to:
    python -m pip install --upgrade pip
    pip install -r requirements.txt

Documenting like this because I don't think the workflows are covered by version control.

**Review by smf541 — APPROVED** [review]

**Pull request merged**